Tile collision updates, restored SaveData ability, reset player positions to default

Axanery opened the pull request.

Axanery committed:
Updated tile collisions to fit new Hatch functions, restored SaveData.json ability, reset temporary player positions

aknetk commented on Scripts/OpenMania/Stage/Common/Ring.hsl:
```diff
@@ -218,21 +218,21 @@ class Ring {
         if (this.XSpeed < 0.0 && ( sensor = TileCollision.Line(this.X, this.Y, 3, halfW, this.PlaneIndex, -1) )) {
-            if (sensor["Collided"]) {
+            if (this.SensorCollided) {
                 this.X = sensor["X"] + halfW;
```
Might wanna fix the `sensor["X"]` and `sensor["Y"]` stuff on this line 222, 229, 236, 245, and wherever else you can Find `sensor["` just in case

aknetk commented on Scripts/OpenMania/Other/TileCollisionDebugger.hsl:
```diff
@@ -36,25 +36,25 @@ class TileCollisionDebugger {
-            if (sensor["Collided"]) {
+            if (this.SensorCollided) {
                 Draw.SetBlendColor(0x00FF00, 1.0);
                 this.DrawArrow(sensor["X"], sensor["Y"], sensor["Angle"], 8);
```
Same as the other comment

aknetk reviewed (changes requested):
Couple changes needed before The m e r g e

Axanery committed:
Updated Sensors

Updated the sensors in both Ring and TileCollisionDebugger, no idea why this took me so long, sorry.

Axanery commented:
I finally fixed those sensors. I have no idea why I never bothered to fix it until just now, sorry. There shouldn't be any more instances of sensor["X"] and the like, they've been replaced.

aknetk commented on Scripts/OpenMania/Stage/Common/Ring.hsl:
```diff
@@ -217,32 +217,32 @@ class Ring {
+        if (this.YSpeed < 0.0 && ( sensor = TileCollision.Line(this.X, this.Y, 2, halfH, this.PlaneIndex, -1, this) )) {
             if (this.SensorCollided) {
-                this.Y = sensor["Y"] + halfH;
+                this.Y = this.SensorYd + halfH;
```
Is this a typo? the extra "d" at the end of "this.SensorYd" I mean

aknetk reviewed (changes requested):
Just a small change

Axanery committed:
Fixed typo

Axanery commented:
Gah, of course there was a typo. My fault. Fixed and pushed.
